### DISCUSSION TITLE
Update Vite Ruby installer to target Vite 6?

### DISCUSSION THREAD

**Discussion by mattbrictson** [ElMassimo/vite_ruby#518] [Q&A] [answered]
Vite 6 was released this past week, and I've been upgrading a couple of my Rails + Vite Ruby apps to use `vite@^6.0.0`. So far it has been a smooth upgrade with no problems I've noticed.

Currently, when installing Vite Ruby on a brand new code base, the Vite Ruby installer explicitly uses `vite@^5.0.0`.

https://github.com/ElMassimo/vite_ruby/blob/b0d1812efec6a27e474c206874746718fcc7f613/vite_ruby/lib/vite_ruby/version.rb#L7

I wonder if should we update this to v6? Are there some compatibility issues I should be looking out for?

**Comment by ElMassimo** [comment, marked as answer]
Hi Matt!

Most apps would require little to no changes when upgrading from Vite 5 to Vite 6.

Apps using SSR might require additional work, given that Vite 6 has different internals (Environment API).

I typically wait until I run a new major for a while in production, as well as keeping an eye for any patches that are released shortly after,  before updating the default in Vite Ruby.

**Reply by jrochkind** [reply]
Thoughts on current timelines? 

I'm actually not sure if we can currently use vite 6 as an opt-in with vite ruby?

(I am not sure if I need/want vite 6 at all, I'm just having vite problems I can't figure out with trying to get sass source maps, so one thing I look at is if my dependencies are up to date).

**Comment by brendon** [comment]
I'm wondering this myself. Is it worth holding off?

**Reply by ElMassimo** [reply]
I'll be likely upgrading the default to Vite 6 in the upcoming weeks.

**Reply by brendon** [reply]
Nice :) Thanks @ElMassimo.
